transformers/ArbitrumPollingTransformer.js

```diff
@@ -47,13 +55,43 @@ const handlers = {
       optionIdInt = info.event.params.optionId.toNumber();
     }
 
+    let delegateContractAddress;
+
+    try {
+      // If we are on the L2 testnet, use the Goerli deployment of Vote Delegate Factory.
+      const vdfAddress =
+        layer2ChainMap[services.networkState.networkName.chainId];
+
+      // The provider needs to be connected to the same L1 network where the delegate contract was created.
+      const provider = new ethers.providers.JsonRpcProvider(
+        process.env.VL_CHAIN_HOST
+      );
+      const delegateFactoryContract = new ethers.Contract(
+        vdfAddress,
+        vdfAbi,
+        provider
+      );
+      delegateContractAddress = await delegateFactoryContract.delegates(
+        info.event.params.voter
+      );
+    } catch (e) {
+      logger.error(
+        `There was an error trying to find the delegate contract address for ${info.event.params.voter.toLowerCase()}, not Inserting 'Voted' event. ${e}`
```

Comment:
hmm, yeah. The RetryProvider is the one that should be accessible at services.providers, although I know you were having some issues with that earlier. Maybe upgrading ethers in the spock package would help since the version used there is Kris's fork from 2019.